assets/src/edit-story/components/panels/test/alignment.js

```diff
@@ -0,0 +1,239 @@
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { fireEvent } from '@testing-library/react';
+
+/**
+ * Internal dependencies
+ */
+import Alignment from '../alignment';
+import { renderPanel } from './_utils';
+
+jest.mock('../../../elements');
+
+describe('Panels/Alignment', () => {
+  let defaultElement, defaultText, defaultImage, fillElement;
+  const horizontalDistributionButtonLabel = 'Horizontal Distribution';
+  const verticalDistributionButtonLabel = 'Vertical Distribution';
+  const justifyLeftButtonLabel = 'Justify Left';
+  const justifyCenterButtonLabel = 'Justify Center';
+  const justifyRightButtonLabel = 'Justify Right';
+  const justifyTopButtonLabel = 'Justify Top';
+  const justifyMiddleButtonLabel = 'Justify Middle';
+  const justifyBottomButtonLabel = 'Justify Bottom';
+
+  beforeEach(() => {
+    defaultElement = {
+      id: '1',
+      isBackground: false,
+      x: 10,
+      y: 10,
+      width: 100,
+      height: 80,
+      isFill: false,
+      rotationAngle: 0,
+    };
+    defaultText = {
+      ...defaultElement,
+      type: 'text',
+    };
+    defaultImage = {
+      ...defaultElement,
+      type: 'image',
+    };
+    fillElement = {
+      ...defaultElement,
+      isFill: true,
+    };
+  });
+
+  function renderAlignment(...args) {
+    return renderPanel(Alignment, ...args);
+  }
+
+  it('should render <Alignment /> panel', () => {
+    const { getByLabelText } = renderAlignment([defaultElement]);
+    const element = getByLabelText('Alignment Panel');
+    expect(element).toBeDefined();
+  });
+
+  describe('single selection', () => {
+    it('should render disabled distribution buttons', () => {
+      const { getByLabelText, pushUpdate } = renderAlignment([defaultElement]);
+      const horizontalDistributionButton = getByLabelText(
+        horizontalDistributionButtonLabel
+      );
+      const verticalDistributionButton = getByLabelText(
+        verticalDistributionButtonLabel
+      );
+      expect(horizontalDistributionButton).toBeDisabled();
+      expect(verticalDistributionButton).toBeDisabled();
+
+      fireEvent.click(horizontalDistributionButton);
+      expect(pushUpdate).toHaveBeenCalledTimes(0);
+
+      fireEvent.click(verticalDistributionButton);
+      expect(pushUpdate).toHaveBeenCalledTimes(0);
+    });
+
+    it('should render align and justify buttons for default element', () => {
+      const { getByLabelText } = renderAlignment([defaultElement]);
+      const justifyLeftButton = getByLabelText(justifyLeftButtonLabel);
+      const justifyCenterButton = getByLabelText(justifyCenterButtonLabel);
+      const justifyRightButton = getByLabelText(justifyRightButtonLabel);
+      const justifyTopButton = getByLabelText(justifyTopButtonLabel);
+      const justifyMiddleButton = getByLabelText(justifyMiddleButtonLabel);
+      const justifyBottomButton = getByLabelText(justifyBottomButtonLabel);
```

Comment:
In these cases it might help to move this to the `renderAlignment` function and just return an object of all those buttons - you'll need them for most tests anyway. It'll reduce the complexity of the tests.

Reply:
Yes, moved them to renderAlignment